internal/server/users.go

```diff
@@ -0,0 +1,155 @@
+package server
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+
+	"golang.org/x/crypto/bcrypt"
+	validator "gopkg.in/go-playground/validator.v9"
+
+	ihttp "github.com/Pigmice2733/peregrine-backend/internal/http"
+	"github.com/Pigmice2733/peregrine-backend/internal/store"
+	jwt "github.com/dgrijalva/jwt-go"
+)
+
+type contextKey string
+
+const (
+	keyRolesContext   contextKey = "pigmice_roles"
+	keySubjectContext contextKey = "pigmice_subject"
+)
+
+type baseUser struct {
+	Username string `json:"username" validate:"gte=4,lte=32"`
```

Comment:
Why are there different limits here than in store?

Reply:
There aren't any in store, are there?

Reply:
Clarified on slack. Good point! Will fix the migrations.